infra/Pos/Network/Types.hs

```diff
@@ -198,6 +203,21 @@ topologyNodeType TopologyP2P{}         = NodeEdge
 topologyNodeType TopologyTraditional{} = NodeCore
 topologyNodeType TopologyAuxx{}        = NodeEdge
 
+-- | Type class which encapsulates something that has 'NodeType'.
+class HasNodeType ctx where
+    -- | Extract 'NodeType' from a context. It's not a 'Lens', because
+    -- the intended usage is that context has topology and it's not
+    -- possible to 'set' node type in topology.
+    getNodeType :: ctx -> NodeType
+
+-- | Implementation of 'getNodeType' for something that has
```

Comment:
"that has ..." `NetworkConfig` i suppose.

Reply:
Oops